core/modules/system/system.admin.inc

```diff
@@ -322,30 +337,41 @@ function system_theme_default() {
     if (!empty($themes[$theme])) {
       // Enable the theme if it is currently disabled.
       if (empty($themes[$theme]->status)) {
-       theme_enable(array($theme));
+        theme_enable(array($theme));
       }
       // Set the default theme.
       config_set('system.core', 'theme_default', $theme);
 
       // Rebuild the menu. This duplicates the menu_rebuild() in theme_enable().
       // However, modules must know the current default theme in order to use
       // this information in hook_menu() or hook_menu_alter() implementations,
-      // and doing the config_set() before the theme_enable() could result
-      // in a race condition where the theme is default but not enabled.
+      // and doing the config_set() before the theme_enable() could result in a
+      // race condition where the theme is default but not enabled.
       menu_rebuild();
 
-      // The status message depends on whether an admin theme is currently in use:
-      // a value of 0 means the admin theme is set to be the default theme.
+      // The status message depends on whether an admin theme is currently in
+      // use: a value of 0 means the admin theme is set to be the default theme.
       $admin_theme = config_get('system.core', 'admin_theme');
+
+      // If the admin theme is set to a different theme than the one the user
+      // has just switched to, show an info message to explain why.
       if ($admin_theme && $admin_theme != $theme) {
-        backdrop_set_message(t('Please note that the administration theme is still set to the %admin_theme theme; consequently, the theme on this page remains unchanged. All non-administrative sections of the site, however, will show the selected %selected_theme theme by default.', array(
+        backdrop_set_message(t('%theme is now the default theme for all non-administrative pages.', array('%theme' => $themes[$theme]->info['name'])));
+        backdrop_set_message(t('The administration theme is still set to the %admin_theme theme. <a href="@change_admin_theme_link">Change the administration theme', array(
```

Comment:
IMO we should just remove this second message. This message makes it sound as though you _should_ change the admin theme, even though that's an uncommon task.

Reply:
The `The administration theme is still set to ...` bit is still required, in order to let people understand that the fact that they are not seeing the new theme applied is normal/expected. ...I see how the `Change the administration theme` link may make it seem as what you describe.

We need to either come up with a better wording, or remove that link (and unfortunately assume that people know that they need to the bottom of the page to change that, if they want).

Reply:
...I have removed the `Change the administration theme` link from the info message, and made the `administration theme` bit a link to the settings at the bottom of the page instead 😉